### DISCUSSION TITLE
Q: How ghidra recovers the type of variable?

### DISCUSSION THREAD

**Discussion by GA-create** [NationalSecurityAgency/ghidra#6928] [Q&A]
Hello, sorry to bother you guys, I am new on ghidra. I would like to ask how ghidra recovers the type of a variable, for example ghidra determines that a stack variable is of type long. Could you please tell me in which module ghdria implements this algorithm and the references for this algorithm.
Thank you very much.

**Comment by hippietrail** [comment]
I believe it uses the type information in the instructions and address modes. If it accesses the same number of bits as in a `long`, it will set it to a `long`. This won't work 100% of the time but will work most of the time. Similar when it comes to signed vs unsigned though there may be fewers hints for this.

Anyway that's how I would do it if I were implementing an RE tool.
